reactioncommerce/reaction-next-starterkit
# Update checkout to support multiple payments

Impact: **breaking**
Type: **feature**

Blocked from merging by https://github.com/reactioncommerce/reaction-component-library/pull/388

Requires https://github.com/reactioncommerce/reaction/pull/4908 for the connected API changes.

## Changes
Wires up the new components introduced in https://github.com/reactioncommerce/reaction-component-library/pull/388. NOTE: They are temporarily copied into starter kit. No need to review the component files here because they are identical to what is in the component library PR. After that is released, this PR will be updated to reference the four new components from the library.

Also updates the checkout and checkout complete pages for all of the breaking data changes introduced by https://github.com/reactioncommerce/reaction/pull/4908.

## Breaking changes
The app is updated to expect the API responses in the breaking API PR: introduced by https://github.com/reactioncommerce/reaction/pull/4908 Placing an order will no longer work against APIs that don't have those changes.

## Testing
1. Verify that you can place an order with a single credit card payment as before.
2. Remove the example IOU payment method from `paymentMethods` array and verify that the checkout looks and works pretty much as it did before this PR.
3. Try placing an order paying with an IOU, specifying no amount. It should place the order paying for the full amount with IOU
4. Try placing an order paying with an IOU for a partial amount. You should see the partial payment appear and the payment form should remain open. Add a second IOU or credit card payment for the remainder. Verify that the order is placed.
5. Try paying with a payment method that you have disabled on the server side. Verify that you get an error message when placing the order.
6. Verify that orders with multiple payments look correct on the order completion page.
7. With the single-payment and multiple-payment orders, thoroughly test the order summary panel in the operator UI. Try variations on approving, capturing, and refunding payments, and canceling the whole order. All of this code has changed or has been moved.
8. Verify that bulk order operations in the operator UI work properly with multi-payment orders.
